LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/JoinRequestMessageHandler.cs

```diff
@@ -184,30 +187,38 @@ async Task ProcessJoinRequestAsync(LoRaRequest request)
                 }
                 else
                 {
-                    Logger.Log(devEUI, $"twin RX2 datarate values are not within acceptable values", LogLevel.Error);
+                    Logger.Log(devEUI, $"twin RX2 datarate value are not within acceptable values", LogLevel.Error);
                 }
 
-                if (loRaDevice.DesiredRX1DROffset >= 0 && loRaDevice.DesiredRX1DROffset < request.Region.RX1DROffsetTable.GetUpperBound(1))
+                if (request.Region.IsValidRX1DROffset(loRaDevice.DesiredRX1DROffset))
                 {
                     var rx1droffset = (byte)(loRaDevice.DesiredRX1DROffset << 4);
                     dlSettings[0] = (byte)(dlSettings[0] + rx1droffset);
                 }
                 else
                 {
-                    Logger.Log(devEUI, $"twin Rx1 offset datarate values are not within acceptable values", LogLevel.Error);
+                    Logger.Log(devEUI, $"twin Rx1 offset datarate value are not within acceptable values", LogLevel.Error);
                 }
 
-                var joinAccept = this.CreateJoinAcceptDownlinkMessage(
-                    netId,
-                    loRaDevice.AppKey,
-                    devAddr,
-                    appNonceBytes,
-                    datr,
-                    freq,
-                    tmst,
-                    devEUI,
-                    dlSettings,
-                    loRaDevice.DesiredRXDelay);
+                ushort rxDelay = 0;
+                if (request.Region.IsValidRXDelay(loRaDevice.DesiredRXDelay))
+                {
+                    rxDelay = loRaDevice.DesiredRXDelay;
+                }
+                else
+                {
+                    Logger.Log(devEUI, $"twin RX Delay value are not within acceptable values", LogLevel.Error);
+                }
+
+                var loRaPayloadJoinAccept = new LoRaTools.LoRaMessage.LoRaPayloadJoinAccept(
+               LoRaTools.Utils.ConversionHelper.ByteArrayToString(netId), // NETID 0 / 1 is default test
```

Comment:
indent is wrong here.

Reply:
Shouldn't Stylecop detect this? :/

Reply:
this is odd should stylecop detect this for sure....